packages/neo-one-client/src/__tests__/provider/neoone/JSONRPCClient.test.js

```diff
@@ -0,0 +1,161 @@
+/* @flow */
+import BigNumber from 'bignumber.js';
+
+import { keys, transactions } from '@neo-one/client-core/src/__data__';
+
+import JSONRPCClient from '../../../provider/neoone/JSONRPCClient';
+import { RelayTransactionError } from '../../../errors';
+
+jest.mock('isomorphic-fetch');
+
+describe('JSONRPCClient', () => {
+  const expected = '10';
+  const jsonProvider = { request: jest.fn(() => Promise.resolve(expected)) };
+  const provider = new JSONRPCClient(jsonProvider);
+  const testCases = [
+    {
+      method: 'getAccount',
+      args: [keys[0].address],
+    },
+    {
+      method: 'getAsset',
+      args: [transactions.register.hash],
+    },
+    {
+      method: 'getBlock',
+      args: [transactions.register.hash],
+    },
+    {
+      method: 'getBlock',
+      args: [0],
+    },
+    {
+      method: 'getBestBlockHash',
+      args: [],
+    },
+    {
+      method: 'getBlockCount',
+      args: [],
+    },
+    {
+      method: 'getContract',
+      args: [keys[0].scriptHash],
+    },
+    {
+      method: 'getMemPool',
+      args: [],
+    },
+    {
+      method: 'getTransaction',
+      args: [transactions.register.hash],
+    },
+    {
+      method: 'getStorageItem',
+      args: [keys[0].scriptHash, Buffer.from('123', 'hex')],
+    },
+    {
+      method: 'getUnspentOutput',
+      args: [
+        {
+          hash: transactions.register.hash,
+          index: 0,
+        },
+      ],
+    },
+    {
+      method: 'testInvokeRaw',
+      args: [Buffer.from('123', 'hex')],
+    },
+    {
+      method: 'sendTransactionRaw',
+      args: [''],
+    },
+    {
+      method: 'relayTransaction',
+      args: [''],
+    },
+    {
+      method: 'getOutput',
+      args: [
+        {
+          hash: transactions.register.hash,
+          index: 0,
+        },
+      ],
+    },
+    {
+      method: 'getClaimAmount',
+      args: [
+        {
+          hash: transactions.register.hash,
+          index: 0,
+        },
+      ],
+      result: new BigNumber(expected),
+    },
+    {
+      method: 'getAllStorage',
+      args: [keys[0].scriptHash],
+    },
+    {
+      method: 'testInvocation',
+      args: [Buffer.alloc(1, 0)],
+    },
+    {
+      method: 'getTransactionReceipt',
+      args: [keys[0].scriptHash],
+    },
+    {
+      method: 'getTransactionReceipt',
+      args: [keys[0].scriptHash],
```

Comment:
Should also test the case where options is non-null (so `watchTimeout` is defined in the snapshot, see line 183 in the snapshot file)